adagrams/game.py

```diff
@@ -1,11 +1,125 @@
+
+import random
+
+
 def draw_letters():
-    pass
+    # array_of_strings = [ ]
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+    letter_pool = []
+    for letter, count in LETTER_POOL.items():
+        letter_pool.extend([letter] * count)
+    
+    hand = []
+    
+    for _ in range(10):
+        random_index = random.randint(0, len(letter_pool) - 1)
+        hand.append(letter_pool[random_index])
+        letter_pool.pop(random_index)
+    
+    return hand
+
+
+
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    word = word.upper()
+    letter_bank_remaining = letter_bank.copy()
+
+    for letter in word:
+        if letter in letter_bank_remaining:
+            letter_bank_remaining.remove(letter)
+        else:
+            return False
+    return True
+
+    
+
 
 def score_word(word):
-    pass
+
+    # Define the letter scores
+    score_chart = {
```

Comment:
If we were going to expand on this project, this dictionary would be a great candidate for global variable status!